pkg/artifactory/resource_artifactory_watch.go

```diff
@@ -0,0 +1,654 @@
+			},
+			"repository_paths": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"include_patterns": {
+							Type:     schema.TypeSet,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Optional: true,
+						},
+						"exclude_patterns": {
+							Type:     schema.TypeSet,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Optional: true,
+						},
+					},
+				},
+			},
+			"all_builds": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"bin_mgr_id": {
+							Type:     schema.TypeString,
+							Default:  "default",
+							Optional: true,
+						},
+						"include_patterns": {
+							Type:     schema.TypeSet,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Optional: true,
+						},
+						"exclude_patterns": {
+							Type:     schema.TypeSet,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Optional: true,
+						},
+					},
+				},
+				MaxItems:      1,
+				ConflictsWith: []string{"build"},
+			},
+			"build": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"bin_mgr_id": {
+							Type:     schema.TypeString,
+							Default:  "default",
+							Optional: true,
+						},
+					},
+				},
+				ConflictsWith: []string{"all_builds"},
+			},
+			"policy": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceWatchCreate(d *schema.ResourceData, m interface{}) error {
+	client := m.(*ArtClient).XrayClient
+
+	if client == nil {
+		return fmt.Errorf("you must specify the xray_url in the provider to manage a watch")
+	}
+
+	params, err := unpackWatch(d, m)
+	if err != nil {
+		return err
+	}
+
+	_, err = client.CreateWatch(*params)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(params.Name)
+
+	return nil
+}
+
+func resourceWatchRead(d *schema.ResourceData, m interface{}) error {
+	client := m.(*ArtClient).XrayClient
+
+	if client == nil {
+		return fmt.Errorf("you must specify the xray_url in the provider to manage a watch")
+	}
+
+	watch, resp, err := client.GetWatch(d.Id())
+
+	if resp.StatusCode == http.StatusNotFound {
+		d.SetId("")
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	return packWatch(watch, d)
+}
+
+func resourceWatchUpdate(d *schema.ResourceData, m interface{}) error {
+	client := m.(*ArtClient).XrayClient
+
+	if client == nil {
+		return fmt.Errorf("you must specify the xray_url in the provider to manage a watch")
```

Comment:
But, that's not what you're actually checking for - you're checking for nil, but assuming the reason for a `nil` is because the url was missing. Asserting client isn't `nil` is fine, I guess. But the error message has to reflect the issue

Reply:
I've updated the error message.